src/Microsoft.AspNetCore.HttpOverrides/IPNetwork.cs

```diff
@@ -1,12 +1,245 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Globalization;
 using System.Net;
+using System.Net.Sockets;
+using System.Collections.Immutable;
 
 namespace Microsoft.AspNetCore.HttpOverrides
 {
     public class IPNetwork
     {
+        private static readonly char[] PrefixLengthSeparators = new char[] { '/' };
+
+        /// <summary>
+        /// All IPv4 addresses 0.0.0.0/0
+        /// </summary>
+        public static readonly IPNetwork AllIPv4 = Parse("0.0.0.0/0");
+
+        /// <summary>
+        /// IPv4 localhost range 127.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4Loopback = Parse("127.0.0.1/8");
+
+        /// <summary>
+        /// IPv4 link-local range 169.254.0.0/16 (255.255.0.0)
+        /// </summary>
+        public static readonly IPNetwork IPv4LinkLocal = Parse("169.254.0.0/16");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 10.0.0.0/8 (255.0.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private10 = Parse("10.0.0.0/8");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 172.16.0.0/12 (255.240.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private172 = Parse("172.16.0.0/12");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 range 192.168.0.0/16 (255.255.0.0)
+        /// </summary>
+        /// <seealso cref="RFC1918" />
+        public static readonly IPNetwork IPv4Private192 = Parse("192.168.0.0/16");
+
+        /// <summary>
+        /// IPv4 Multicast range 224.0.0.0/4
+        /// </summary>
+        public static readonly IPNetwork IPv4Multicast = Parse("224.0.0.0/4");
+
+        /// <summary>
+        /// RFC1918 Private IPv4 Addresses 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16
+        /// </summary>
+        /// <seealso cref="IPv4PrivateNetworks" />
+        public static readonly ImmutableArray<IPNetwork> RFC1918 = ImmutableArray.Create(IPv4Private10, IPv4Private172, IPv4Private192);
```

Comment:
The name is precise, but not descriptive. Give it a descriptive name and put the RFC in the doc comments.